Source/WebKit/Platform/IPC/ArgumentCoders.h

```diff
@@ -75,33 +75,44 @@ template<typename T, size_t Extent> struct ArgumentCoder<Span<T, Extent>> {
     template<typename Encoder>
     static void encode(Encoder& encoder, const Span<T, Extent>& span)
     {
-        if constexpr (Extent == WTF::dynamic_extent)
-            encoder << static_cast<uint64_t>(span.size());
-        encoder.encodeFixedLengthData(reinterpret_cast<const uint8_t*>(span.data()), span.size() * sizeof(T), alignof(T));
+        if constexpr (Extent == WTF::dynamic_extent) {
+            auto size = static_cast<uint64_t>(span.size());
```

Comment:
I think this warrants a change in behavior where we allow per-platform encodings as per:
I believe original idea was that all types were sized, presumably to enforce that messages are of same length in all platforms.
I don't see this being a requirement anywhere, and I don't think it is a benefit.

So going forward with Span changes, we could just encode size_t, with the benefit that the code is automatically correct, not needing the isInBounds.

If you want this commit to reflect the original code from this perspective, feel free to ignore the suggestion and we can make it later.

Reply:
This makes sense. I'll apply it in a separate PR to spans and other types.